tests/tc_ufcs_completions/file.d

```diff
@@ -7,4 +6,9 @@ void main()
 {
 	auto foo = Foo();
 	foo.
+	foo.ufcsSelf.
```

Comment:
this looks like it should auto-complete `foo.foo.ufcsSelf.` (because this is a valid multi-line statement)

in that case, no autocomplete should be expected, but it looks like the tests test for something here. I don't think just looking at the last function name to see what it returns is a great idea. When there are multiple possible overloads we don't take them into account, templates aren't resolved (and it won't really be possible to implement templates with this approach)

Because of this the current approach quite obviously gives us invalid completion results here, just guessing on the previous method name. This is quite a showstopper I think because it will confuse users why DCD autocompletes functions and the compiler rejects them. (especially bad for new users)

Reply:
Oh I didn't know about multi-line statement, so you suggest that I need to validate the whole chain right?